### ISSUE TITLE
KML Export Inserts Bogus Points in Polygons

### ISSUE THREAD

**Issue by sarcanon** [isawnyu/pleiades-gazetteer#225]
First noticed by @gabrielbodard:

Summary: The KML dumps appear to insert points into polygon Location representations that don't belong.

Test case: See the KML for the site Isca https://pleiades.stoa.org/places/79532. This site has three Locations associated with it. One of them (the fortress' boundary walls) is a polygon. That polygon, consisting of five points, is correctly represented on the Pleaides web site. However, the KML representation of that polygon includes a sixth point .

`<coordinates>-2.955010,51.607700,0.000000 -2.956878,51.608059,nan -2.960240,51.610610,0.000000 -2.956280,51.613320,0.000000 -2.951030,51.610460,0.000000 -2.955010,51.607700,0.000000</coordinates>`

Note the point in the second position (-2.956878,51.608059,nan). First, that point is not a member of the Polygon. It is in fact the Point for another Location associated with this Place, the Amphitheatre https://pleiades.stoa.org/places/79532/roman-amphitheatre. Secondly, note the NaN, which is probably explained by trying to cast a Point (with no altitude value) as a member of a Polygon.

**Comment by cguardia** [comment]
This took me some time to analyze, but I believe the export is not inserting bogus points. What happens is that the dump is just for places, not individual locations, so the code tries to get the smallest convex Polygon containing all the points in the combined locations for a place. There are adapters inside adapters here, but the final result is a call to the convex_hull method of the multipoint geometry (see shapely library), which returns the points for this enclosing polygon. This is the value that is indexed for the place and used by the kml dump.

In other words, this is the expected behavior according to the code.

**Comment by paregorios** [comment]
Interestingly, something seems to have changed here. The KML file referenced in the original ticket submission now contains the following:

```
<coordinates>-2.955010,51.607700,0.000000 -2.960240,51.610610,0.000000 -2.956280,51.613320,0.000000 -2.951030,51.610460,0.000000 -2.955010,51.607700,0.000000</coordinates>
```

The amphitheatre seems no longer to be pulled into the calculation or the polygon. 

Some other bugfix (or a version upgrade to something underneath like shapely?) must have fixed this. 

@sarcanon can you verify to your satisfaction that this is no longer happening?

**Comment by sarcanon** [comment]
Well, technically, it's not my ticket, i.e., I'm not the original reporter of the bug. This was Bodard's issue, but I helped him track it down, so I submitted the ticket on his behalf. I'm not entirely certain how he produced the export that surfaced the anomalous results, so I don't know how to reproduce the problem.

**Comment by paregorios** [comment]
Thanks @sarcanon 

@gabrielbodard would you please have a look at the latest KML and see if this problem persists?

**Comment by cguardia** [comment]
@paregorios are you sure you generated the kml using the dump-scripts/kml script from the buildout? That's the one that has the polygon code. If you append 'kml' to a place it will show up correctly because it outputs each placemark separately.

**Comment by paregorios** [comment]
@cguardia I wasn't aware that we are using different code to produce kml serializations of individual places and for the dumps. I'll take this ticket back for further review.

**Comment by paregorios** [comment]
This ticket appears to have arisen from the same misunderstanding of the Pleiades KML dump (as distinct from the per-place KML serializations) that I held. @cguardia is right: the KML dump is working as designed in this case: it provides a single, bounding polygon (i.e., convex hull) for each place that contains a location with polygon geometry. If there are multiple subordinate locations, then the bounding polygon surrounds all of these.

I note that the particular example cited above does not conform to current editorial guidance. The amphitheater should be split out as its own, separate place. 

Closing "won't fix".
